tasks/system_probe.py

```diff
@@ -683,6 +683,7 @@ def build_security_probe_ebpf_files(ctx, build_dir, parallel_build=True, kernel_
 
     security_flags = get_ebpf_build_flags(kernel_release=kernel_release)
     security_flags.append(f"-I{security_agent_c_dir}")
+    security_flags.append("-DUSE_RING_BUFFER=1")
```

Comment:
I am a bit afraid that, at some point, we forget to specify USE_RING_BUFFER=1 and, if the required kernel headers are not present, the feature will simply not work. When we runtime compile, it makes sense to not compile a feature that we won't be present, but here we precompile programs for every supported kernel release and that's expected that the kernel used to compile does not have the same features than the ones it will run on.

Reply:
When building the version that is shipped to customers, I agree.

Unfortunately, this would force a two VM model on our local development, where we have to build system-probe in a VM that is 5.8+ (or has access to 5.8+ kernel headers), and then transferring the binary and all the ancillary files to the VM we want to test in.

I think adding a safety check to CI would be sufficient to ensure you never ship something to customers without ring buffer support.

Reply:
If you VM is based on Ubuntu Focal, a `apt install linux-headers-5.14.0-1024-oem` is enough to have the proper kernel headers.

Reply:
Yeah, when troubleshooting issues we would like the ability to build on much older VMs, that won't have 5.8 kernel headers available.